core/src/test/java/org/apache/iceberg/TestDefaultValuesParsingAndUnParsing.java

```diff
@@ -0,0 +1,116 @@
+    this.type = type;
+    this.defaultValue = defaultValue;
+  }
+
+  @Parameterized.Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][] {
+        {Types.BooleanType.get(), "null"},
+        {Types.BooleanType.get(), "true"},
+        {Types.IntegerType.get(), "1"},
+        {Types.LongType.get(), "9999999"},
+        {Types.FloatType.get(), "1.23"},
+        {Types.DoubleType.get(), "123.456"},
+        {Types.DateType.get(), "\"2007-12-03\""},
+        {Types.TimeType.get(), "\"10:15:30\""},
+        {Types.TimestampType.withoutZone(), "\"2007-12-03T10:15:30\""},
+        {Types.TimestampType.withZone(), "\"2007-12-03T10:15:30+01:00\""},
+        {Types.StringType.get(), "\"foo\""},
+        {Types.UUIDType.get(), "\"eb26bdb1-a1d8-4aa6-990e-da940875492c\""},
+        {Types.FixedType.ofLength(2), "\"111f\""},
+        {Types.BinaryType.get(), "\"0000ff\""},
+        {Types.DecimalType.of(9, 2), "123.45"},
+        {Types.ListType.ofOptional(1, Types.IntegerType.get()), "[1, 2, 3]"},
+        {Types.MapType.ofOptional(1, 2, Types.IntegerType.get(), Types.StringType.get()),
+         "{\"keys\": [1, 2], \"values\": [\"foo\", \"bar\"]}"},
+        {Types.StructType.of(
+            required(1, "f1", Types.IntegerType.get(), "doc"),
+            optional(2, "f2", Types.StringType.get(), "doc")),
+         "{\"1\": 1, \"2\": \"bar\"}"},
+        // deeply nested complex types
+        {Types.ListType.ofOptional(1, Types.StructType.of(
+            required(1, "f1", Types.IntegerType.get(), "doc"),
+            optional(2, "f2", Types.StringType.get(), "doc"))), "[{\"1\": 1, \"2\": \"bar\"}, {\"1\": 2, \"2\": " +
+             "\"foo\"}]"},
+        {Types.MapType.ofOptional(1, 2, Types.IntegerType.get(), Types.StructType.of(
+            required(1, "f1", Types.IntegerType.get(), "doc"),
+            optional(2, "f2", Types.StringType.get(), "doc"))),
+         "{\"keys\": [1, 2], \"values\": [{\"1\": 1, \"2\": \"bar\"}, {\"1\": 2, \"2\": \"foo\"}]}"},
+        {Types.StructType.of(
+            required(1, "f1", Types.StructType.of(
+                optional(2, "ff1", Types.IntegerType.get(), "doc"),
+                optional(3, "ff2", Types.StringType.get(), "doc")), "doc"),
+            optional(4, "f2", Types.StructType.of(
+                optional(5, "ff1", Types.StringType.get(), "doc"),
+                optional(6, "ff2", Types.IntegerType.get(), "doc")), "doc")),
+         "{\"1\": {\"2\": 1, \"3\": \"bar\"}, \"4\": {\"5\": \"bar\", \"6\": 1}}"},
+        });
+  }
+
+  // serialize to json and deserialize back should return the same result
+  static String defaultValueParseAndUnParseRoundTrip(Type type, String defaultValue) throws IOException {
+    Object javaDefaultValue = DefaultValueParser.fromJson(type, defaultValue);
+    return DefaultValueParser.toJson(type, javaDefaultValue);
+  }
+
+  @Test
+  public void testTypeWithDefaultValue() throws IOException {
```

Comment:
Instead of having a parameterized test suite with just one test, can you refactor this and embed the list of types you want to test?

Then you can combine all of the default values tests into a single suite.

Reply:
I made them into a single test class.